conda_build/noarch_python.py

```diff
@@ -24,7 +26,7 @@ def rewrite_script(fn):
     directory after it passes some sanity checks for noarch pacakges"""
 
     # Load and check the source file for not being a binary
-    src = join(config.build_prefix, 'bin', fn)
+    src = join(config.build_prefix, 'Scripts' if ISWIN else 'bin', fn)
```

Comment:
My thought was to prevent things that are executables from being made into noarch packages.  So, packages like 7zip or Qt or curl or ... - just those kinds of things.  File extensions don't make sense for those on *nix, but they do on Windows.

I doubt that people will try to package windows exe's for mac/linux, but they are likely to try to make something noarch for the sake of using only one msvc runtime, possibly.  That won't work, and we should try to help them understand why, ideally with an error message here?

There are ways to make EXE-only packages not depend on a python version (hard-code the compiler, and give it the correct runtime dependency) - and this is sort of what people would want as "no-arch" - but it's different from noarch stuff here.